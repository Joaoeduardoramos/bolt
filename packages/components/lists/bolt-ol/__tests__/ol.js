import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

describe('<bolt-ol> Component', async () => {
  afterAll(async () => {
    await stopTwigRenderer();
  }, 15000);

  test('basic usage with attributes', async () => {
    const results = await renderTwig('@bolt-components-ol/ol.twig', {
      attributes: {
        'data-attr': 'some attribute',
        onclick: "location.href='https://pega.com'",
      },
      items: [
        'Do not include any data or information in your posts that are confidential!',
        'Apply basic practices for collaborative work.',
        'Be honest, respectful, trustworthy and helpful.',
        'Answer questions authoritatively and concisely. Avoid cluttering discussions with noise.',
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('with nested <bolt-ul> list compiles', async () => {
    const results = await renderTwigString(`
      {% include '@bolt-components-ol/ol.twig' with {
        items: [
          'Do not include any data or information in your posts that are confidential!',
          'Apply basic practices for collaborative work.',
          'Be honest, respectful, trustworthy and helpful.',
          'Answer questions authoritatively and concisely. Avoid cluttering discussions with noise.',
          [
            'Answer questions authoritatively and concisely.',
            include('@bolt-components-ul/ul.twig', {
              items: [
                'Item 1',
                'Item 2',
                'Item 3',
                'Item 4'
              ]
            })
          ]
        ]
      } %}
    `);
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('with nested <bolt-ol> list compiles', async () => {
    const results = await renderTwigString(`
      {% include '@bolt-components-ol/ol.twig' with {
        items: [
          'Do not include any data or information in your posts that are confidential!',
          'Apply basic practices for collaborative work.',
          'Be honest, respectful, trustworthy and helpful.',
          'Answer questions authoritatively and concisely. Avoid cluttering discussions with noise.',
          [
            'Answer questions authoritatively and concisely.',
            include('@bolt-components-ol/ol.twig', {
              items: [
                'Item 1',
                'Item 2',
                'Item 3',
                'Item 4'
              ]
            })
          ]
        ]
      } %}
    `);
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
