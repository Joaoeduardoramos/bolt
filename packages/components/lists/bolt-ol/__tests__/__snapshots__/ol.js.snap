// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-image> <bolt-card> 1`] = `

<bolt-ol>
  <replace-with-grandchildren>
    <ol class="some-class another-class c-bolt-ol"
        onclick="location.href=&#039;https://pega.com&#039;"
    >
      <bolt-li>
        <replace-with-grandchildren>
          <li class="c-bolt-li c-bolt-li--ol-item ">
            Do not include any data or information in your posts that are confidential!
          </li>
        </replace-with-grandchildren>
      </bolt-li>
      <bolt-li>
        <replace-with-grandchildren>
          <li class="c-bolt-li c-bolt-li--ol-item ">
            Apply basic practices for collaborative work.
          </li>
        </replace-with-grandchildren>
      </bolt-li>
      <bolt-li>
        <replace-with-grandchildren>
          <li class="c-bolt-li c-bolt-li--ol-item ">
            Be honest, respectful, trustworthy and helpful.
          </li>
        </replace-with-grandchildren>
      </bolt-li>
      <bolt-li>
        <replace-with-grandchildren>
          <li class="c-bolt-li c-bolt-li--ol-item c-bolt-li--last-item">
            Answer questions authoritatively and concisely. Avoid cluttering discussions with noise.
          </li>
        </replace-with-grandchildren>
      </bolt-li>
    </ol>
  </replace-with-grandchildren>
</bolt-ol>

`;
